.gitlab-ci.yml

```diff
@@ -130,21 +130,95 @@ publish:s3:apt-repo:
     - aws s3 cp --recursive s3://$S3_BUCKET_NAME/$S3_BUCKET_REPO_PATH/dists dists
     - aws s3 cp --recursive s3://$S3_BUCKET_NAME/$S3_BUCKET_REPO_PATH/pool pool
   script:
-    # If build from final tag, include in stable
-    - if echo "${MENDER_VERSION}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+$'; then
-    -   for change_file in $(ls ${CI_PROJECT_DIR}/output/mender-client_*.changes); do
-    -     reprepro --keepunreferencedfiles include stable $change_file
-    -   done
+    # Bash function to check if the string is a tag
+    - |
+      function is_tag () {
+        version=$1
+        echo "${version}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+$'
+        return $?
+      }
+    # Bash function to compare two semantic versions, returns: 0 (=), 1(>) or 2(<)
+    # Source: https://stackoverflow.com/a/4025065
+    - |
+      function vercomp () {
+        if [[ $1 == $2 ]]
+        then
+          return 0
+        fi
+        local IFS=.
+        local i ver1=($1) ver2=($2)
+        # fill empty fields in ver1 with zeros
+        for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+        do
+          ver1[i]=0
+        done
+        for ((i=0; i<${#ver1[@]}; i++))
+        do
+          if [[ -z ${ver2[i]} ]]
+          then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+          fi
+          if ((10#${ver1[i]} > 10#${ver2[i]}))
+          then
+            return 1
+          fi
+          if ((10#${ver1[i]} < 10#${ver2[i]}))
+          then
+            return 2
+          fi
+        done
+        return 0
+      }
+    # Bash function to check if the given version for the given package is newer
+    # than the current version of that package in the APT repo
+    - |
+      function is_newer_version() {
+        package_name=$1
+        package_version=$2
+
+        current_version=$(reprepro --architecture amd64 --list-format '${version}' list stable ${package_name})
+        if [ -z "${current_version}" ]; then
+          # Package doesn't exist, so we assume is "newer"
+          return 0
+        fi
+        current_version_semver=$(echo $current_version | cut -d"-" -f1 )
+
+        vercomp $package_version $current_version_semver
+        if [ $? -eq 1 ]; then
+          return 0
```

Comment:
Just double checking: It still works if you use reprepro to upload a package with the same version as the newest, right?

Reply:
Oops, no, I misread this, it's the opposite of what I said, which will of course work. Never mind.